pkg/parsers/parsers.go

```diff
@@ -187,7 +187,7 @@ func ParseMap(params map[string]interface{}, parent string, index int, queryResp
 		case reflect.Map:
 			m := value.(map[string]interface{})
 
-			result, err := ParseMap(m, keyname, index, queryRespMap)
+			result, err := ParseMap(m, keyname, -1, queryRespMap)
```

Comment:
what about when we do need to pass the index recursively down?

Reply:
Do you have any examples? I'm not seeing where we would need to pass the index down more than one level. If I'm parsing a value, would I need to know what index I'm at in the grandparent?

Reply:
mmm i cannot come up with one either. it seems to be doing what you did in other places too. i think it is fine!